src/cli/index.ts

```diff
@@ -0,0 +1,153 @@
+
+  try {
+    const pkgContent: string = fs.readFileSync(pkgPath, 'utf-8')
+    pkg = JSON.parse(pkgContent)
+  }
+  catch (error) {
+    console.log(c.red('Could not find -'), c.dim('package.json'))
+    process.exit(1)
+  }
+
+  const eslintLastVersion = await execSync(`npm show eslint version --quite`)
+  const eslintConfigLastVersion = await execSync(`npm show @antfu/eslint-config version --quite`)
+
+  const eslintVersion: string = eslintLastVersion?.toString()?.trim()
+  const eslintConfigVersion: string = eslintConfigLastVersion?.toString()?.trim()
+
+  pkg.type = 'module'
+
+  const pkgDep = ['dependencies', 'devDependencies']
+
+  pkgDep.forEach((depType) => {
+    if (pkg[depType]) {
+      Object.keys(pkg[depType]).forEach((key) => {
+        if (key.includes('eslint') || key.includes('prettier'))
+          delete pkg[depType][key]
+      })
+    }
+  })
+
+  pkg.devDependencies['@antfu/eslint-config'] = eslintConfigVersion
+  pkg.devDependencies.eslint = eslintVersion
+
+  try {
+    fs.writeFileSync(pkgPath, JSON.stringify(pkg, null, 2))
+    console.log(c.cyan('Updated -'), c.dim('package.json'))
+  }
+  catch (error) {
+    console.log(c.red('Failed to update -'), c.dim('package.json'))
+    process.exit(1)
+  }
+  // End update package.json
+
+  // Update eslint files
+  const eslintConfigPath: string = path.join(cwd, 'eslint.config.js')
+  const eslintIgnorePath: string = path.join(cwd, '.eslintignore')
+  let eslintIgnoreLines: string[] = []
+
+  if (fs.existsSync(eslintIgnorePath)) {
+    const eslintIgnoreContent: string = fs.readFileSync(eslintIgnorePath, 'utf-8')
+    eslintIgnoreLines = eslintIgnoreContent.split('\n')
+  }
+
+  const eslintConfigContent = `
+import antfu from '@antfu/eslint-config';
+
+export default antfu({
+  ${eslintIgnoreLines.length
+  ? `ignores: [
+    ${eslintIgnoreLines?.map(line => `'${line}'`).join(',\n')}
+  ],`
+ : ''}
+});`
+
+  const files = fs.readdirSync(cwd)
+  files.forEach((file) => {
+    if (file.includes('eslint') || file.includes('prettier'))
+      fs.unlinkSync(file)
+  })
+
+  try {
+    fs.writeFileSync(eslintConfigPath, eslintConfigContent)
+    console.log(c.cyan('Created -'), c.dim('eslint.config.js'))
+  }
+  catch (error) {
+    console.log(c.red('Failed to create -'), c.dim('eslint.config.js'))
+  }
+  // End update eslint files
+
+  // Update .vscode/settings.json
+  const dotVscodePath: string = path.join(cwd, '.vscode')
+  const settingsPath: string = path.join(dotVscodePath, 'settings.json')
+
+  if (!fs.existsSync(dotVscodePath))
+    fs.mkdirSync(dotVscodePath)
+
+  if (!fs.existsSync(settingsPath))
+    fs.writeFileSync(settingsPath, '{}')
+
+  let settings: Record<string, any> = {}
+  try {
+    const settingsContent: string = fs.readFileSync(settingsPath, 'utf8')
+    const settingsJsonWithoutComments = settingsContent.replace(/\/\/.*|\/\*[\s\S]*?\*\//g, '')
+    settings = JSON.parse(settingsJsonWithoutComments)
```

Comment:
I think we won't want to remove all comments. Maybe instead of parsing, we insert the text before the last `}`. Leaving the formating to the ESLint config.

Reply:
Its quite a difficult situation, in the sense that I cant parse it because of the presence of comments and I cant overwrite the fields as such, because I read it as a string line. I tried to do as you said and take into account all the problematic situations, but im not sure that such an implementation is okay 🥴